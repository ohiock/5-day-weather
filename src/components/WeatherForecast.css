.forecastTitle {
  font-size: 4.5vw;
}

.forecastDay {
  padding: 1.5vw;
}

.icon {
  margin: 0 0 1vw 0;
}

.temp {
  margin: 0 0 1vw 0;

  font-size: 3vw;
}

.condition {
  font-weight: bold;
}

.background0 {
  background: rgba(0, 0, 0, 0.1);
}

.background1 {
  background: rgba(0, 0, 0, 0.15);
}

.background2 {
  background: rgba(0, 0, 0, 0.2);
}

.background3 {
  background: rgba(0, 0, 0, 0.25);
}

.background4 {
  background: rgba(0, 0, 0, 0.3);
}

.goBackContainer {
  padding: 2vw;

  font-size: 2vw;
  font-weight: bold;
}

.goBackButton {
  cursor: pointer;
}

@media (max-width: 768px) {
  .forecastTitle {
    font-size: 7.5vw;
  }

  .forecastDay {
    padding: 3vw;

    font-size: 4vw;
  }

  .temp {
    margin: 0 0 2vw 0;

    font-size: 6vw;
  }

  .goBackContainer {
    padding: 4vw;

    font-size: 6vw;
  }
}
