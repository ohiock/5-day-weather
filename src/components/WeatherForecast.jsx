import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CSSModules from 'react-css-modules';
import Grid from 'react-bootstrap/lib/Grid';
import Row from 'react-bootstrap/lib/Row';
import Col from 'react-bootstrap/lib/Col';
import moment from 'moment';

import ForecastHeader from './ForecastHeader';
import ForecastIcon from './ForecastIcon';

import sharedStyles from '../Shared.css';
import styles from './WeatherForecast.css';

const propTypes = {
  show: PropTypes.bool.isRequired,
  weatherForecast: PropTypes.object.isRequired,
};

class WeatherForecast extends Component {
  constructor(props) {
    super(props);

    this.state = {

    };
  }

  render() {
    return (
      <Grid styleName={this.props.show ? 'show' : 'hide'}>
        <Row>
          <Col>
            <div styleName="text-center">
              <h1 styleName="forecast-title">5 Day Weather Forecast for {this.props.weatherForecast.city}</h1>
            </div>
            <Row>
              {
                this.props.weatherForecast.days
                  .map((day, i) => (
                    <Col mdOffset={i === 0 ? 1 : 0} md={2} key={day.date} styleName={`forecast-day background-${i}`}>
                      <ForecastHeader
                        weekDay={moment.unix(day.date).format('dddd')}
                        today={moment().format('dddd')}
                        tomorrow={moment().add(1, 'days').format('dddd')} />
                      <div styleName="text-center icon"><ForecastIcon icon={day.icon} /></div>
                      <div styleName="temp">{ `${Math.round(day.temp)}Â°` }</div>
                      <div styleName="condition">{ day.condition }</div>
                      <div styleName="description">{ day.description }</div>
                    </Col>
                  ))
              }
            </Row>
          </Col>
        </Row>
      </Grid>
    );
  }
}

WeatherForecast.propTypes = propTypes;

export default CSSModules(WeatherForecast, Object.assign(sharedStyles, styles), { allowMultiple: true });
