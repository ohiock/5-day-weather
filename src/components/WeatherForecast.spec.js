import React from 'react';
import renderer from 'react-test-renderer';

import WeatherForecast from './WeatherForecast';

it('renders correctly', () => {
  const forecast = {
    city: '',
    days: []
  }

  const tree = renderer.create(
    <WeatherForecast weatherForecast={forecast} goBackCallback={() => true} show={true} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
