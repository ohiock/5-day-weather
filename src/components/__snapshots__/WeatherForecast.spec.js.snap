// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <div
        className={undefined}
      >
        <h1
          className={undefined}
        >
          5 Day Weather Forecast
          <br />
           for 
          
        </h1>
      </div>
      <div
        className="row"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className=""
    >
      <div
        className="undefined undefined"
        onClick={[Function]}
      >
        <span
          className={undefined}
        >
          Search Again
        </span>
      </div>
    </div>
  </div>
</div>
`;
